PhysicsTools/Heppy/python/analyzers/objects/JetAnalyzer.py

```diff
@@ -119,12 +119,39 @@ def process(self, event):
         
 	##Sort Jets by pT 
         allJets.sort(key = lambda j : j.pt(), reverse = True)
+        
+        leptons = []
+        if hasattr(event, 'selectedLeptons'):
+            leptons = [ l for l in event.selectedLeptons if l.pt() > self.lepPtMin and self.lepSelCut(l) ]
+        if self.cfg_ana.cleanJetsFromTaus and hasattr(event, 'selectedTaus'):
+            leptons = leptons[:] + event.selectedTaus
+        if self.cfg_ana.cleanJetsFromIsoTracks and hasattr(event, 'selectedIsoCleanTrack'):
+            leptons = leptons[:] + event.selectedIsoCleanTrack
 	## Apply jet selection
         self.jets = []
         self.jetsFailId = []
         self.jetsAllNoID = []
         self.jetsIdOnly = []
         for jet in allJets:
+            #Check if lepton and jet have overlapping PF candidates 
+            leps_with_overlaps = []
+            for i in range(jet.numberOfSourceCandidatePtrs()):
+                p1 = jet.sourceCandidatePtr(i) #Ptr<Candidate> p1
+                if not p1.isAvailable():
+                    continue
+                for lep in leptons:
+                    for j in range(lep.numberOfSourceCandidatePtrs()):
+                        p2 = lep.sourceCandidatePtr(j)
+                        if not p2.isAvailable():
+                            continue
+                        #for some reason, this does not work.
+                        #has_overlaps = p1.refCore() == p2.refCore() and p1.key() == p2.key()
```

Comment:
I have written this for converting a ref into a tuple that can be compared for identity
https://github.com/gpetruc/cmg-cmssw/blob/CMGTools-from-CMSSW_7_4_X/CMGTools/TTHAnalysis/python/analyzers/ttHSVAnalyzer.py#L133-L135
maybe it can be elevated to some common place in Heppy?